planet/clients/features.py

```diff
@@ -0,0 +1,183 @@
+        ...         cl = sess.client('features')
+        ...         # use client here
+        ...
+        >>> asyncio.run(main())
+        ```
+    """
+
+    def __init__(self,
+                 session: Session,
+                 base_url: Optional[str] = None) -> None:
+        """
+        Parameters:
+            session: Open session connected to server.
+            base_url: The base URL to use. Defaults to production features
+                API base url.
+        """
+        self._session = session
+
+        self._base_url = base_url or BASE_URL
+        if self._base_url.endswith('/'):
+            self._base_url = self._base_url[:-1]
+
+    async def list_collections(self) -> AsyncIterator[dict]:
+
+        class _CollectionsPager(Paged):
+            """Navigates pages of messages about collections."""
+            ITEMS_KEY = 'collections'
+            LINKS_KEY = 'links'
+
+            def _next_link(self, page):
+                next_link = False
+
+                # TODO: Build this into Paged.__next_link directly, other (unimplemented) APIs have a similar page structure
+                for link in page[self.LINKS_KEY]:
+                    if "rel" in link and link["rel"] == self.NEXT_KEY:
+                        next_link = link["href"]
+                        LOGGER.debug(f'next: {next_link}')
+                if not next_link:
+                    LOGGER.debug('end of the pages')
+                return next_link
+
+        url = f'{self._base_url}/collections'
+
+        try:
+            response = await self._session.request(method='GET', url=url)
+            async for col in _CollectionsPager(response,
+                                               self._session.request):
+                yield col
+        # Forward APIError. We don't strictly need this clause, but it
+        # makes our intent clear.
+        except APIError:
+            raise
+        except ClientError:  # pragma: no cover
+            raise
+
+    async def list_features(
+            self,
+            collection_id,
+            limit: int = 10,
+            bbox: Optional[list[float]] = None,
+            datetime: Optional[str] = None) -> AsyncIterator[dict]:
+
+        class _FeaturesPager(Paged):
+            """Navigates pages of messages about features."""
+            ITEMS_KEY = 'features'
+            LINKS_KEY = 'links'
+
+            def _next_link(self, page):
+                next_link = False
+
+                #TODO: Build this into Paged.__next_link directly, other (unimplemented) APIs have a similar page structure
+                for link in page[self.LINKS_KEY]:
+                    if "rel" in link and link["rel"] == self.NEXT_KEY:
+                        next_link = link["href"]
+                        LOGGER.debug(f'next: {next_link}')
+                if not next_link:
+                    LOGGER.debug('end of the pages')
+                return next_link
+
+        params = {}
```

Comment:
```suggestion
        params: list[str, Any] = {}
```

This is failing the mypy check since we're adding multiple types into this dict.